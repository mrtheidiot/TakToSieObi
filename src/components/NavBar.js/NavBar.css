.nav-wrapper {
  height: vh;
}

.nav-section-wrapper {
  width: 100%;
  position: fixed;
  background-color: white;
  box-shadow: 0px 10px 50px grey;
}

.nav-secion__position {
  display: flex;
  justify-content: space-between;
  width: 90%;
  margin: auto;
  align-items: center;
}

.nav-section {
  display: flex;
  justify-content: space-between;
  width: 70%;
}

.bor-bot {
  border-bottom: solid green 1px;
}

@media (max-width: 992px) {
  .nav-section {
    display: none;
  }
}

@media (min-width: 992px) {
  .nav-section-menuicon {
    display: none;
  }
}

.nav-section-menuicon {
  height: 100%;
}

.menuIcon {
  margin: auto 0; 
}