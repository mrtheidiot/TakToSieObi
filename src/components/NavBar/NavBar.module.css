.wrapper {
    position: fixed;
    top: 0;
    z-index: 100;
    width: 100%;
    height: 100px;
    background-color: #f9faef;
    display: flex;
    justify-content: space-between;
    box-shadow: rgba(17, 17, 26, 0.1) 0px 4px 16px, rgba(17, 17, 26, 0.1) 0px 8px 24px, rgba(17, 17, 26, 0.1) 0px 16px 56px;
}

.logo {
    display: flex;
}

.logo img {
    height: 80%;
    margin: auto 0 auto 20px;
}

.navBar_mobile {
    display: flex;
    margin: auto 20px auto 0;
}

.navBar_wide {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    margin-right: 20px;
}

@media (min-width: 868px) {
    .navBar_mobile {
        display: none;
    }
}

@media (max-width: 868px) {
    .navBar_wide {
        display: none;
    }
}
