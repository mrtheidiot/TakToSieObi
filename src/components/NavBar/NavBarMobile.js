import Toggler from "../../UI/Toggler/Toggler";
import { useState } from "react";
import { NavLink } from "react-router-dom";

import Logo from "./../../assets/Logo1.svg";
import classes from "./NavBarMobile.module.css";

// NavBarMobile shows on vewport width that is less than 868px in a form of overlay

const Overlay = ({
  navData,
  isLoggedIn,
  setEdititngModeHandler,
  onClose,
  editMode,
}) => {
  return (
    <div className={classes.overlay}>
      <div className={classes.overlay_animations}>
        <i className="fas fa-times fa-4x" onClick={onClose} />
        <NavLink to="/" className={classes.overlay_logo} onClick={onClose}>
          <img src={Logo} />
        </NavLink>
        <div className={classes.overlay_list}>
          {navData.map((link) => (
            <NavLink onClick={onClose} to={link.to}>
              {link.title}
            </NavLink>
          ))}
        </div>
        {isLoggedIn && (
          <Toggler onChange={setEdititngModeHandler} current={editMode} />
        )}
      </div>
    </div>
  );
};

const NavBarMobile = (props) => {
  const [showOverlay, setShowOverlay] = useState(false);

  const toggleNavBarOverlay = () => {
    setShowOverlay((prev) => !prev);
  };

  return (
    <>
      <i
        className="fas fa-bars fa-3x"
        onClick={toggleNavBarOverlay}
        data-testid="hamburger"
      />
      {showOverlay && <Overlay {...props} onClose={toggleNavBarOverlay} />}
    </>
  );
};

export default NavBarMobile;
