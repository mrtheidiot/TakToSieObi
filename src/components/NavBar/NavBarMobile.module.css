.overlay {
    position: fixed;
    z-index: 1000;
    top: 0;
    left: 0;
    /* -webkit-animation: fadeInFromNone 0.5s ease-out;
    -moz-animation: fadeInFromNone 0.5s ease-out;
    -o-animation: fadeInFromNone 0.5s ease-out;
    animation: fadeInFromNone 0.5s ease-out; */
}

 .overlay_animations {
    position: absolute;
    left: -100vw;
    background: black;
    -webkit-animation: slide 0.5s forwards;
    animation: slide 0.5s forwards;
    color: white;
    width: 100vw;
    height: 100vh;
    text-align: center;
}

.overlay_animations i {
    transition: ease 0.3s rotate;
    float: right;
    margin: 10px 20px;
  }
  
  .overlay_animations i:hover {
    rotate: 90deg;
}  

.overlay_list {
    margin-top: 30px;
    display: flex;
    align-items: center;
    flex-direction: column;
}

.overlay_list a {
    text-decoration: none;
    color: white;
    font-family: 'Bebas Neue', cursive;
    font-size: 3rem;
    transition: ease 0.3s opacity;
    margin: 3px 0;
}

.overlay_list a:hover {
    opacity: 0.5;
}

.overlay_logo {
    height: 20vh;
    width: 100%;
    justify-content: center;
    display: flex;
    align-items: center;
}

.overlay_logo img {
    height: 20vh;
}

@-webkit-keyframes fadeInFromNone {
from {
    opacity: 0;
}

to {
    opacity: 1;
}
}

@keyframes fadeInFromNone {
    from {
        opacity: 0;
    }

    to {
        opacity: 1;
    }
} 

@-webkit-keyframes slide {
    100% {
        left: 0;
    }
    }
    
    @keyframes slide {
    100% {
        left: 0;
    }
}