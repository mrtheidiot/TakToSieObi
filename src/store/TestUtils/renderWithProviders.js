import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { setupStore } from "./store";

// customer renderer which preloads the store with content for testing components using redux

function renderWithProviders(
  ui,
  { preloadedState = {}, store = setupStore(preloadedState) } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>
      {children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper }) };
}


export default renderWithProviders;
