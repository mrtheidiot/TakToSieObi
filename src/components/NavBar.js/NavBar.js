import MenuItem from "../MenuItem/MenuItem";
import Logo from "../MenuItem/Logo";
import "./NavBar.css";

const NavBar = (props) => {
  return (
    <div className="nav-wrapper">
      <div className="nav-section-wrapper">
        <div className="nav-secion__position">
          <div className="nav-logo">
            <Logo />
          </div>
          <div className="nav-section">
            <MenuItem
              name={"TRENING OBEDIENCE"}
              destination={"/treningi"}
              dropdown={props.trainingTitles}
            />
            <MenuItem
              name={"ASORTYMENT"}
              destination={"/asortyment"}
              dropdown={props.itemTitles}
            />
            <MenuItem name={"JA I MOJE PSY"} destination={"/omnie"} />
            <MenuItem name={"CENNIK"} destination={"/cennik"} />
            <MenuItem name={"KALENDARZ"} destination={"/kalendarz"} />
            <MenuItem name={"KONTAKT"} destination={"/kontakt"} />
          </div>
          <div
            className="nav-section-menuicon"
            onClick={props.handleOverlayMenu}
          >
            <div className="menuIcon">
              <i class="fas fa-bars fa-3x"></i>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default NavBar;
