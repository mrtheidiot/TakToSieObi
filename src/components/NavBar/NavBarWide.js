import Toggler from "../../UI/Toggler/Toggler";
import { NavLink, Link } from "react-router-dom";

import "./../../stylus/dist/NavBar.css";

// NavBarWide shows on vewport width that is more than 868px in a form of overlay
// this component uses the Stylus css preprocessor

const NavBarWide = ({
  navData,
  isLoggedIn,
  setEdititngModeHandler,
  editMode,
}) => {
  const navItemsList = navData.map((link) => (
    <nav className="navLink_navElement" key={link.id}>
      <NavLink to={`${link.to}`} activeClassName="navLink_active">
        {link.title}
      </NavLink>
      {link.dropdown && (
        <ul className="dropdown_menu dropdown_animated">
          <li>
            {link.dropdown.map((item) => (
              <Link key={item.id} to={item.link}>
                {item.title}
              </Link>
            ))}
          </li>
        </ul>
      )}
    </nav>
  ));

  return (
    <>
      {navItemsList}
      {isLoggedIn && (
        <Toggler onChange={setEdititngModeHandler} current={editMode} />
      )}
    </>
  );
};

export default NavBarWide;
