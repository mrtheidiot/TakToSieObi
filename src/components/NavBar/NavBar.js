import NavBarWide from "./NavBarWide";
import NavBarMobile from "./NavBarMobile";
import { useSelector, useDispatch } from "react-redux";
import { uiActions } from "../../store/ui-slice";
import { Link } from "react-router-dom";

import Logo from "./../../assets/Logo1.svg";
import classes from "./NavBar.module.css";

// Navigation Bar always shows the logo, the links or mobile modal are dependent on the viewport width
// Once admin is logged in - a toogler shows and allows the editMode to be turned on
// editMode allows admin to perform CRUD actions on the page
// logging in is inteded to be only for admin and it's possible to access via manually going to '/login'

const NavBar = () => {
  const dispatch = useDispatch();
  const courses = useSelector((state) => state.courses.coursesContent);
  const editMode = useSelector((state) => state.ui.editMode);

  const setEdititngModeHandler = (e) => {
    dispatch(uiActions.setEditMode(e.target.checked));
  };

  const isLoggedIn = window.localStorage.getItem("isLoggedIn");

  const navData = [
    { id: 1, title: "TRENINGI OBEDIENCE", to: "/treningi", dropdown: courses, pre: "/treningi/" },
    { id: 3, title: "WYDARZENIA", to: "/wydarzenia" },
    { id: 4, title: "JA I MOJE PSY", to: "/omnie" },
    { id: 5, title: "KONTAKT", to: "/kontakt" },
  ];

  return (
    <div className={classes.wrapper}>
      <Link to="/" className={classes.logo}>
        <img src={Logo} alt="logo TakToSieObi" />
      </Link>
      <nav className={classes.navBar_wide}>
        <NavBarWide
          navData={navData}
          isLoggedIn={isLoggedIn}
          setEdititngModeHandler={setEdititngModeHandler}
          editMode={editMode}
        />
      </nav>
      <nav className={classes.navBar_mobile}>
        <NavBarMobile
          navData={navData}
          isLoggedIn={isLoggedIn}
          setEdititngModeHandler={setEdititngModeHandler}
          editMode={editMode}
        />
      </nav>
    </div>
  );
};

export default NavBar;
