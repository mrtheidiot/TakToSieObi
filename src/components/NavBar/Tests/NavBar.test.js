import NavBar from "./../NavBar";
import { BrowserRouter as Router } from "react-router-dom";

import renderWithProviders from "./../../../store/TestUtils/renderWithProviders";

it("matches the NavBar snapshot", () => {
  const initialState = {
    ui: { editMode: false },
  };

  const {
    container: { firstChild },
  } = renderWithProviders(
    <Router>
      <NavBar />
    </Router>,
    {
      preloadedState: initialState,
    }
  );

  expect(firstChild).toMatchSnapshot();
});
