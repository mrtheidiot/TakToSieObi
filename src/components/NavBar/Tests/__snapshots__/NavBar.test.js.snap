// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the NavBar snapshot 1`] = `
<div
  class="wrapper"
>
  <a
    class="logo"
    href="/"
  >
    <img
      alt="logo TakToSieObi"
      src="Logo1.svg"
    />
  </a>
  <nav
    class="navBar_wide"
  >
    <nav
      class="navLink_navElement"
    >
      <a
        href="/treningi"
      >
        TRENINGI OBEDIENCE
      </a>
      <ul
        class="dropdown_menu dropdown_animated"
      >
        <li />
      </ul>
    </nav>
    <nav
      class="navLink_navElement"
    >
      <a
        href="/wydarzenia"
      >
        WYDARZENIA
      </a>
    </nav>
    <nav
      class="navLink_navElement"
    >
      <a
        href="/omnie"
      >
        JA I MOJE PSY
      </a>
    </nav>
    <nav
      class="navLink_navElement"
    >
      <a
        href="/kontakt"
      >
        KONTAKT
      </a>
    </nav>
  </nav>
  <nav
    class="navBar_mobile"
  >
    <i
      class="fas fa-bars fa-3x"
      data-testid="hamburger"
    />
  </nav>
</div>
`;
